src/Runner.cpp

```diff
@@ -22,40 +25,75 @@ Runner::Runner(const vector<ITransformation *> *transformations,
         n_transformations(n_transformations),
         gen(new mt19937(SEED)) {}
 
-void Runner::createOutputFolders(int num_files, char * input_files[], const string& output_path) {
-    fs::path output_dir(output_path);
-    if (!fs::exists(output_dir)) {
-        fs::create_directory(output_dir);
-    }
+void Runner::createDescriptionFile(int num_files, char ** files,
+                                   const string& output_path, const string &description) {
     for (size_t file_index = 0; file_index < num_files; ++file_index) {
-        fs::path current_file_path(input_files[file_index]);
-        fs::path transformations_path = fs::path(output_path) / current_file_path.stem();
-        if (!fs::exists(transformations_path)) {
+        fs::path src_path(files[file_index]);
+        fs::path transformations_path = fs::path(output_path) / src_path.stem();
+        fs::path description_path = transformations_path / "description.txt";
+        ofstream description_stream(description_path, ios_base::app);
+        description_stream << description;
+    }
+}
+
+map<int, char **> Runner::createOutputFolders(int num_files, char * input_files[], const string& output_path) {
+    fs::path output_dir(output_path);
+    fs::create_directory(output_dir);
+
+    map<int, vector<string>> rewritable_paths;
+    map<int, char **> rewritable_cpaths;
```

Comment:
It seems very complicated. Let's discuss, what this function does and what we can optimize here.